presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -335,6 +335,19 @@ public PlanWithProperties visitAggregation(AggregationNode node, StreamPreferred
                 // !isPresent() indicates the property was satisfied completely
                 preGroupedSymbols = groupingKeys;
             }
+            else {
+                // Enable segmented aggregation if the prefix of the sorted-by columns is a subset of the group by column.
+                // Use the longest prefix of the sorted-by columns that satisfy the requirement to maximize the segment aggregation's preGroupedSymbols
+                for (int i = child.getProperties().getLocalProperties().size(); i > 0; i--) {
+                    List<LocalProperty<VariableReferenceExpression>> sortedByKeysPrefix = child.getProperties().getLocalProperties().subList(0, i);
```

Comment:
getLocalProperties() may contain non-sortedBy property. We need to filter it.
Besides, groupby property could also work. It doesn't have to be sorted by property. We can support only sortedBy property for now and add a TODO to support groupby property later.